### PR TITLE
[Feature][Transform] add JsonPath transform

### PR DESCRIPTION
**Pull request by FuYouJ**
## related issue
https://github.com/apache/seatunnel/issues/5633

### Check list

* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [x] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [x] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -128,6 +128,7 @@
         <testcontainer.version>1.17.6</testcontainer.version>
         <spotless.version>2.29.0</spotless.version>
         <jsqlparser.version>4.5</jsqlparser.version>
+        <jsonpath.version>2.8.0</jsonpath.version>
```
Comment (ruanwenjun) [inline pom.xml]: There already exist `2.7.0`, it's not a good idea to used multiple version `json-path`.

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (FuYouJ) [reply pom.xml]: not found a dependency for 2.7.0 yet.Could you point out where it is thank you

Comment (ruanwenjun) [reply pom.xml]: https://github.com/apache/seatunnel/blob/a15b79f5ac8b49e969a64601d9b9436003d0d0d3/seatunnel-connectors-v2/connector-http/connector-http-base/pom.xml#L32-L37

**Comment by FuYouJ** [comment]
@EricJoy2048 could you review this pr,thx

**Comment by FuYouJ** [comment]
> @EricJoy2048 could you review this pr,thx

jsonpath depend on
- net.minidev:json-smart:2.4.7
- net.minidev:accessors-smart:2.4.7
- org.ow2.asm:asm:9.1

File: seatunnel-dist/release-docs/LICENSE [hunk 2]
```diff
@@ -275,6 +275,7 @@ The text of each license is the standard Apache 2.0 license.
      (Apache-2.0) listenablefuture (com.google.guava:listenablefuture:9999.0-empty-to-avoid-conflict-with-guava https://mvnrepository.com/artifact/com.google.guava/listenablefuture/9999.0-empty-to-avoid-conflict-with-guava)
      (Apache-2.0) accessors-smart (com.google.guava:accessors-smart:2.4.7 - https://mvnrepository.com/artifact/net.minidev/accessors-smart)
      (Apache-2.0) json-smart (net.minidev:json-smart:2.4.7 - https://mvnrepository.com/artifact/net.minidev/json-smart)
+     (Apache-2.0) json-path (com.jayway.jsonpath:json-path:2.7.0 - https://mvnrepository.com/artifact/com.jayway.jsonpath/json-path)
```
Comment (ruanwenjun) [inline seatunnel-dist/release-docs/LICENSE:279]: Also need to announce following dependency in LICENSE
```
listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar
 json-smart-2.4.7.jar
 accessors-smart-2.4.7.jar
 asm-9.1.jar
```

File: seatunnel-dist/release-docs/licenses/LICENSE-jayway-jsonpath.txt [hunk 3]
```diff
@@ -0,0 +1,202 @@
+
```
Comment (ruanwenjun) [inline seatunnel-dist/release-docs/licenses/LICENSE-jayway-jsonpath.txt]: Following dependency is under Apache License 2, so this file is not needed, directly announce 
```
listenablefuture-9999.0-empty-to-avoid-conflict-with-guava
json-smart-2.4.7
accessors-smart-2.4.7
json-path-2.7.0
```
in https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/release-docs/LICENSE

asm-9.1.jar is under BSD 3-Clause you need to add asm license file
https://asm.ow2.io/license.html

**Review by ruanwenjun — DISMISSED** [review]

File: seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/jsonpath/JsonPathTransform.java [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+import java.util.Map;
+import java.util.Set;
+import java.util.function.IntFunction;
+import java.util.stream.IntStream;
+
+import static org.apache.seatunnel.transform.exception.JsonPathTransformErrorCode.INPUT_FIELD_NOT_FOUND;
+import static org.apache.seatunnel.transform.exception.JsonPathTransformErrorCode.JSON_PATH_COMPILE_ERROR;
+
+@Slf4j
+@AutoService(JsonPathTransform.class)
```
Comment (TyrantLucifer) [inline seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/jsonpath/JsonPathTransform.java]: Why not Transform.class?

File: seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/jsonpath/FiledConfig.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+package org.apache.seatunnel.transform.jsonpath;
+
+import java.io.Serializable;
+
+public class FiledConfig implements Serializable {
+    private String path;
+
+    private String srcField;
+
+    private String destField;
```
Comment (TyrantLucifer) [inline seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/jsonpath/FiledConfig.java]: final could be better.

**Review by TyrantLucifer — COMMENTED** [review]
Please add parse array and parse nest object test cases.

**Comment by FuYouJ** [comment]
@ruanwenjun jun✌🏻 please review

**Comment by ruanwenjun** [comment]
CI failed, you need to check if the code has miss something.

**Comment by FuYouJ** [comment]
> CI failed, you need to check if the code has miss something.

well,I finally fixed the problem. now, ci is successed

File: seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/jsonpath/JsonPathTransform.java [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+
+import static org.apache.seatunnel.transform.exception.JsonPathTransformErrorCode.INPUT_FIELD_NOT_FOUND;
+import static org.apache.seatunnel.transform.exception.JsonPathTransformErrorCode.JSON_PATH_COMPILE_ERROR;
+
+@Slf4j
+public class JsonPathTransform extends MultipleFieldOutputTransform {
+
+    public static final String PLUGIN_NAME = "JsonPath";
+    private JsonPathTransformConfig config;
+    private SeaTunnelRowType seaTunnelRowType;
```
Comment (TyrantLucifer) [inline seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/jsonpath/JsonPathTransform.java]: final be better

**Review by TyrantLucifer — DISMISSED** [review]
Please add e2e test cases to parse `array` `nest object`

**Comment by FuYouJ** [comment]
@ruanwenjun @ruanwenjun @EricJoy2048 
I refactored the code reference httpsource and fakesource to make it simpler and more powerful. Please review

**Comment by FuYouJ** [comment]
@ruanwenjun jun✌🏻 please review

**Comment by FuYouJ** [comment]
@ruanwenjun @TyrantLucifer  It's been a long time. I miss you so much(✺ω✺)

**Review by hailin0 — APPROVED** [review]

**Review by Hisoka-X — APPROVED** [review]
Thanks @FuYouJ !

**Comment by FuYouJ** [comment]
> Thanks @FuYouJ !

You're welcome!

**Pull request merged**